web/src/main/java/se/inera/intyg/webcert/web/csintegration/user/CertificateServiceUserHelper.java

```diff
@@ -84,6 +81,15 @@ private boolean hasSubscription(WebCertUser webCertUser) {
         return !webCertUser.getSubscriptionInfo().getCareProvidersMissingSubscription().contains(careProviderId);
     }
 
+    private CertificateServiceUserRole getRole(WebCertUser webCertUser) {
+        final var roles = webCertUser.getRoles();
+        if (roles == null || roles.values().isEmpty()) {
+            throw new IllegalStateException("User has no roles");
+        }
+
+        return convertRole(roles.values().stream().findFirst().orElseThrow().getDesc());
+    }
+
```

Comment:
Now that we use WebCertUser we can also simplify the mapping to role. Consider something like below:

Soon the role of nurse will also exist in AuthoritiesConstants (with this PR: https://github.com/sklintyg/infra/pull/248)

Also it is better to fail fast if we don't recognise the role, so we don't get problems "further in" (read CS).

`
private CertificateServiceUserRole convertRole(String role) {
        switch (role) {
            case AuthoritiesConstants.ROLE_LAKARE:
                return CertificateServiceUserRole.DOCTOR;
            case AuthoritiesConstants.ROLE_PRIVATLAKARE:
                return CertificateServiceUserRole.PRIVATE_DOCTOR;
....
....
....
            default:
                throw new IllegalArgumentException("Dont recognise the role: " + role);
        }
    }
`

Reply:
Sorry, the formatting wasn't good when adding code in comment... :-(